### ISSUE TITLE
False positive "Redundant else" inspection when matching on slices

### ISSUE THREAD

**Issue by dima74** [intellij-rust/intellij-rust#5353]
## Environment

* **IntelliJ Rust plugin version:** 0.2.122.3086-201-nightly
* **Rust toolchain version:** 1.43.0 (4fb7144ed 2020-04-20) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.1.1 (CL-201.7223.86)
* **Operating system:** Linux 5.6.11-arch1-1

## Problem description
![image](https://user-images.githubusercontent.com/6505554/81600290-d1349e80-93e2-11ea-98c9-118ea04ad3cc.png)

## Steps to reproduce
```rust
fn main() {
    let vec = &[1, 2][..];
    let single = if let &[single] = &vec {
        Some(single)
    } else {
        None
    };
    dbg!(single);
}
```
